tests/integration/tests/configure_charm.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""Pre-test configuration of a prometheus-juju-exporter testing model."""
+
+import logging
+from base64 import b64encode
+from pathlib import Path
+
+import yaml
+from zaza import model
+
+from .test_charm import wait_for_model_settle
+
+logger = logging.getLogger(__name__)
+
+JUJU_LOCAL_DATA = Path("/home/ubuntu/.local/share/juju/")
+ACCOUNT_FILE = JUJU_LOCAL_DATA.joinpath("accounts.yaml")
+CONTROLLER_FILE = JUJU_LOCAL_DATA.joinpath("controllers.yaml")
+
+
+def setup_juju_credentials() -> None:
+    """Configure prometheus-juju-exporter with required juju credentials.
+
+    Credentials are pulled form config files of a nested controller deployed by
+    'juju-local' charm during the tests.
+    """
+    controller_units = model.get_units("juju-local")
+    if len(controller_units) < 1:
+        err = "Failed to find 'juju-local' units."
+        logger.error(err)
+        raise RuntimeError(err)
+
+    # Get juju controller credentials
+    juju_controller = controller_units[0].entity_id
+    account_data = model.run_on_unit(juju_controller, f"cat {ACCOUNT_FILE}")
```

Comment:
Why do you need to get `accounts.yaml` or `controllers.yaml` from controller? It's not enough to get the information from local juju client?

Reply:
these credentials are for controller deployed within the test model by `juju-local`. During the tests the exporter collects data from the "nested" controller, not the controller that deploys test bundle

Reply:
Ack, thanks for explanation.